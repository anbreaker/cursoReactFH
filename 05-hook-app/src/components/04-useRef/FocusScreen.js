import React, {useRef} from 'react';

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
    console.log(inputRef);
  };

  return (
    <div className="container">
      <h1> Focus Screen</h1>

      <input ref={inputRef} className="form-control" placeholder="Name" />

      <button className="mt-5 btn btn-outline-primary" onClick={handleClick}>
        input
      </button>
    </div>
  );
};
