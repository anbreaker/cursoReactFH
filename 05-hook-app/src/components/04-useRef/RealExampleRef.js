import React from 'react';
import {MultipleUseHooks} from '../03-examples/MultipleUseHooks';

export const RealExampleRef = () => {
  const [show, setShow] = React.useState(false);

  return (
    <div className="container">
      <h1>Real Example hook Ref</h1>
      <hr />

      {show && <MultipleUseHooks />}

      <button
        className="mt-2 btn btn-primary"
        onClick={() => {
          setShow(!show);
        }}>
        Show / Hide
      </button>
    </div>
  );
};
