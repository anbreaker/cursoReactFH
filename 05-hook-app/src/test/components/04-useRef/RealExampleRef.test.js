import {shallow} from 'enzyme';
import {RealExampleRef} from '../../../components/04-useRef/RealExampleRef';

describe('Pruebas en <RealExampleRef /> ', () => {
  const wrapper = shallow(<RealExampleRef />);
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleUseHooks').exists()).toBe(false);
  });

  test('debe mostrar el componente <MultipleUseHooks /> ', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleUseHooks').exists()).toBe(true);
  });
});
