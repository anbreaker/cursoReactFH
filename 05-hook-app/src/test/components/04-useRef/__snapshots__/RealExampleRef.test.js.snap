// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef />  debe mostrarse correctamente 1`] = `
<div
  className="container"
>
  <h1>
    Real Example hook Ref
  </h1>
  <hr />
  <button
    className="mt-2 btn btn-primary"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
